core/shortcut_items.js

```diff
@@ -63,8 +63,8 @@ const registerDelete = function() {
   const deleteShortcut = {
     name: names.DELETE,
     preconditionFn: function(workspace) {
-      return !workspace.options.readOnly && Blockly.selected &&
-          Blockly.selected.isDeletable();
+      return !workspace.options.readOnly && common.getSelected() &&
+          common.getSelected().isDeletable();
```

Comment:
Not sure if it's OK to use yet, but I observe this would be a perfect place to use the optional chaining operator: `common.getSelected()?.isDeletable()`.

Reply:
Also not sure if this is OK yet so leaving it as-is; agreed this will be nice once we can though!